### DISCUSSION TITLE
Trouble adding custom repo

### DISCUSSION THREAD

**Discussion by domrcrel** [uyuni-project/uyuni#8170] [General]
I'm trying to add this custom repository to Uyuni but it doesn't ever seem to load correctly. I assume the formatting of the URLs makes it misbehave, but how can I resolve this?

I'm looking to add the MongoDB repos, as here:

`https://repo.mongodb.org/apt/ubuntu jammy/mongodb-org/6.0 multiverse`

Normally, you would expect this to load as below:

`https://repo.mongodb.org/apt/ubuntu?uyuni_suite=jammy/mongodb-org/6.0&uyuni_component=multiverse`

However, running a `spacewalk-repo-sync` on the channel I made for it shows as below:

````
ERROR: Download of package list failed.
15:56:22 Repo URL: https://repo.mongodb.org/apt/ubuntu?uyuni_suite=jammy/mongodb-org/6.0&uyuni_component=multiverse
15:56:22     Packages in repo:                 0
15:56:22     No new packages to sync.
````

Similarly, adding `uyuni_arch` also fails.

````
ERROR: Download of package list failed.
16:02:22 Repo URL: https://repo.mongodb.org/apt/ubuntu?uyuni_suite=jammy/mongodb-org/6.0&uyuni_component=multiverse&uyuni_arch=amd64
16:02:22     Packages in repo:                 0
16:02:22     No new packages to sync.
````

**Comment by rjmateus** [comment]
I think you need to add as repo URL the full repo up to the architecture. Something like `https://repo.mongodb.org/apt/ubuntu/dists/jammy/mongodb-org/6.0/multiverse/binary-amd64/`

**Comment by domrcrel** [comment]
That works! Thanks!
